sky/skylet/job_lib.py

```diff
@@ -294,13 +294,25 @@ def get_latest_job_id() -> Optional[int]:
         return job_id
 
 
-def get_job_submitted_or_ended_timestamp_payload(
-        job_id: int, get_ended_time: bool) -> Optional[int]:
+def get_job_submitted_or_ended_timestamp_payload(job_id: int,
+                                                 get_ended_time: bool) -> str:
+    """Get the job submitted/ended timestamp.
+
+    This will only be used by the spot controller, which is ok
+    to use `submitted_at` instead of `start_at`, because the
+    spot job duration need to include both setup and running
+    time and the job will not stay in PENDING state.
+
+    The normal job duration will use `start_at` instead of
+    `submitted_at`, because the job may stay in PENDING if
+    the cluster is busy.
+    """
     field = 'end_at' if get_ended_time else 'submitted_at'
     rows = _CURSOR.execute(f'SELECT {field} FROM jobs WHERE job_id=(?)',
                            (job_id,))
     for (timestamp,) in rows:
         return common_utils.encode_payload(timestamp)
+    return common_utils.encode_payload(None)
```

Comment:
Good catch! Why didn't we trigger this bug?

Reply:
The reason is that in the controller, we always query the `job_id` that exists in the spot table in the normal case.